src/main/java/com/microsoft/azure/functions/annotation/Retry.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.functions.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>
+ * Defines Retry Policy</p>
+ *
+ * @since 1.0.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Retry {
```

Comment:
I have a question what is the system behavior in case the user added this parameters.
  "retry": {
    "strategy": "exponentialBackoff",
     "maxRetryCount": 4,
    "delayInterval": "00:00:05"
  }

Reply:
This will throw error. Please see here for validation logic: https://github.com/Azure/azure-functions-host/blob/5652e9287273a6303c631a68cef902f0e8ab4776/src/WebJobs.Script/Utility.cs#L782-L816

Reply:
Good point. I think an error happens on the build time might be good. Opposite case is the same (Fixed Delay but, it has minimum/maximum interval.)

Reply:
Sounds good, I believe this error is run time, should we put a validation in the plugin for the compile time? 
Also I will suggest to make it clear in the annotation that delayInterval can be used to the fixed_delay.

Reply:
Yes. maven plugin should validate it. I'm fine for both design, however, that's make sense. Now we have two design idea. 

## Have two annotations(Fixed/Exponential) 
Pros:   We can force the parameter structure and reduce maven plugin validation logic.
           Similar user experience as C# 
Cons:  We have two annotation to use.

## Have one annotation (Retry)
Pros:   (Probably) simplify maven plugin logic. 
Cons:  We need to validation logic for maven plugins. 

I'm fine for both. However, lean into two annotations since maven plugin need to validate it as well. What I'm not sure the implementation complexity of the maven-plugin.  What do you think @Flanker32 ?

Reply:
I highly recommend not adding validation logic in the mvn plugin as we do not want to duplicate logic in tooling. It will become hard to keep the logic in sync. IMO, For build time validation, keeping annotations separate would be a better approch.

Reply:
I believe two annotations will be good approach as this will align with C#, and make it much easier to use.

Reply:
Decision by majority. :) I'll do separate it again and go ahead and push it. However, I'll keep the commit log.

Reply:
Made it two.